src/gpgpu-sim/gpu-cache.cc

```diff
@@ -704,16 +704,9 @@ unsigned long long cache_stats::operator()(int access_type, int access_outcome,
                                            bool fail_outcome) const {
   ///
   /// Const accessor into m_stats.
-  ///
   if (fail_outcome) {
-    if (!check_fail_valid(access_type, access_outcome))
```

Comment:
Can you wrap these in #if DEBUG instead of just deleting them?
We used to have a debug precompiled flag - but I can't seem to find it now.

Reply:
Are we settling for moving all these assert checks to debug, or do we still want them used for PR regressions? If we still want to have them for regressions, then we'd just need to have a 3rd build option like the PR suggests. I don't have a strong opinion either way, because it's not immediately clear to me which asserts should only be for debug mode, and which we would still want to enable when there are PRs to the codebase.

Reply:
Debug guards have been added